core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -93,7 +93,7 @@ public DeterministicKey(HDPath hdPath,
                             byte[] chainCode,
                             BigInteger priv,
                             @Nullable DeterministicKey parent) {
-        super(priv, ECKey.publicPointFromPrivate(priv), true);
+        super(priv, new LazyECPoint(ECKey.publicPointFromPrivate(priv), true));
```

Comment:
Why not create a 1-parameter constructor that takes a private key? It could then do the `new LazyECPoint(ECKey.publicPointFromPrivate(priv), true)` internally and simplify the callers.

Reply:
Sorry, I missed that comment earlier. I plan to modernize the DeterrministicKey constructors once we've got a canonical constructor. And this (simple) PR was a precondition for having one.